cluster-autoscaler/capacitybuffer/translators/pod_template_translator.go

```
@@ -20,40 +20,53 @@ import (
+func NewPodTemplateBufferTranslator(client *cbclient.CapacityBufferClient) *podTemplateBufferTranslator {
+	return &podTemplateBufferTranslator{
+		client: client,
+	}
 }
 
 // Translate translates buffers processors into pod capacity.
 func (t *podTemplateBufferTranslator) Translate(buffers []*v1.CapacityBuffer) []error {
 	errors := []error{}
+	var numberOfPods *int32
+	var podTemplateRef *v1.LocalObjectRef
 	for _, buffer := range buffers {
 		if isPodTemplateBasedBuffer(buffer) {
-			podTemplateRef, numberOfPods, err := t.translate(buffer)
+			podTemplateRef = buffer.Spec.PodTemplateRef
+			podTemplate, err := t.client.GetPodTemplate(podTemplateRef.Name)
 			if err != nil {
-				setBufferAsNotReadyForProvisioning(buffer, err.Error())
+				common.SetBufferAsNotReadyForProvisioning(buffer, nil, nil, nil, err)
 				errors = append(errors, err)
-			} else {
-				setBufferAsReadyForProvisioning(buffer, podTemplateRef.Name, numberOfPods)
+				continue
+			}
+			numberOfPods = t.getNumberOfReplicas(buffer)
+			if numberOfPods == nil {
+				common.SetBufferAsNotReadyForProvisioning(buffer, podTemplateRef, &podTemplate.Generation, nil, fmt.Errorf("Failed to get buffer's number of pods"))
+				continue
 			}
+			common.SetBufferAsReadyForProvisioning(buffer, podTemplateRef, &podTemplate.Generation, numberOfPods)
 		}
 	}
 	return errors
 }
 
-func (t *podTemplateBufferTranslator) translate(buffer *v1.CapacityBuffer) (*v1.LocalObjectRef, int32, error) {
-	// Fixed Replicas will be used if both Replicas and Percent are defined
+func (t *podTemplateBufferTranslator) getNumberOfReplicas(buffer *v1.CapacityBuffer) *int32 {
 	if buffer.Spec.Replicas != nil {
-		return buffer.Spec.PodTemplateRef, max(1, int32(*buffer.Spec.Replicas)), nil
+		replicas := max(0, int32(*buffer.Spec.Replicas))
+		return &replicas
 	}
-	return nil, 0, fmt.Errorf("Failed to translate buffer %v, Replicas should have a value when PodTemplateRef is set", buffer.Name)
+	return nil
```

Do we have a validation in place that would check:
- if pod template is used either replicas, limit or both should be set?

It would be best to catch simple validations at admission rather than in the buffer controller loop (fail fast makes it easier for the users to realize problems)

---

No we didn't, added it